src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs

```diff
@@ -113,6 +137,7 @@ public Uri SourceUri
             string apiKey,
             TimeSpan requestTimeout,
             ILogger log,
+            bool useTempApiKey,
```

Comment:
I would expect this to be decided at the end, passing it in provides more places for it to be wrong